protos/protos/public/modeldb/versioning/VersioningService.proto

```diff
@@ -418,6 +418,45 @@ message RevertRepositoryCommitsRequest {
     }
 }
 
+message FindRepositories {
+    repeated uint64 repo_ids = 1;
+    repeated string repo_names = 2;
+    string workspace_name = 3;
+    repeated KeyValueQuery predicates = 4;
+
+    //For pagination
+    int32 page_number = 6;
+    int32 page_limit = 7;
+    //In future requirement
+    //bool ascending = 8;
+    //string sort_key = 9;
+
+    message Response {
+        repeated Repository repositories = 1;
+        int64 total_records = 2;
+    }
+}
+
+message FindRepositoriesBlobs {
+    repeated uint64 repo_ids = 1;
+    repeated string repo_names = 2;
+    string workspace_name = 3;
+    repeated string location_prefix = 4;
+    int32 blob_type = 5;
```

Comment:
Enum so that we know what are the blob types. I also suspect this should be repeated so that I can say "S3 dataset or path dataset".

Reply:
if you want blob_type as `repeated ` field then which Operator is helpful in the query. is it `AND` OR `OR` operator?

Reply:
`OR`. You can't have something being multiple blob types.

Reply:
Ok, make sense.